test/integration/scan_test.go

```diff
@@ -509,3 +507,25 @@ func TestFailedScanWithWrongPreset(t *testing.T) {
 	err, _ := executeCommand(t, args...)
 	assertError(t, err, "scan did not complete successfully")
 }
+
+func TestScanCreateWithSSHKey(t *testing.T) {
+	_ = viper.BindEnv("CX_SCAN_SSH_KEY")
+	sshKey := viper.GetString("CX_SCAN_SSH_KEY")
+
+	filename := "ssh-key-file.txt"
+	_ = ioutil.WriteFile(filename, []byte(sshKey), 0644)
+
+	_, projectName := getRootProject(t)
+
+	args := []string{
+		"scan", "create",
+		flag(params.ProjectName), projectName,
+		flag(params.SourcesFlag), SSHRepo,
+		flag(params.BranchFlag), "main",
+		flag(params.SSHKeyFlag), filename,
+	}
+
+	executeCmdWithTimeOutNilAssertion(t, "Create a scan with ssh-key should pass", 4*time.Minute, args...)
+
+	_ = os.Remove(filename)
```

Comment:
If any of the test steps fail the file won't be removed.
We should do:
`defer func() {
    _ = os.Remove(filename)
}
`
right after the file creation to ensure it gets deleted at the end of the test